File: spring-cloud-contract/java-spring-cloud-contract/app-caller/src/main/java/com/example/javaspringcloudcontract/external/CalleeCaller.java
package com.example.javaspringcloudcontract.external;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(
        name = "calleeCaller",
        url = "${callee.base-url}",
        primary = false
)
public interface CalleeCaller {

    @GetMapping("/api/v1/callee")
    ResponseEntity<String> callCalleeApplication(@RequestParam int num);
}

File: spring-cloud-contract/app-caller/src/main/kotlin/com/example/springcloudcontract/external/CalleeCaller.kt
package com.example.springcloudcontract.external

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient(
    name = "calleeCaller",
    url = "\${callee.base-url}",
    primary = false,
)
interface CalleeCaller {
    @GetMapping("/api/v1/callee")
    fun callCalleeApplication(@RequestParam num: Int): ResponseEntity<String>
}